### PR TITLE
TST: check for best-effort parsing in foreign locales

### PR DESCRIPTION
**Pull request by aarchiba**

### CODE REVIEW

File: numpy/core/tests/test_longdouble.py [hunk 1]
```diff
@@ -80,6 +81,11 @@ def test_fromstring():
                  err_msg="reading '%s'" % s)
 
 
+@in_foreign_locale
+def test_fromstring_best_effort():
+    assert_equal(np.fromstring("1,234", dtype=float, sep=" "),
```
Comment (charris) [inline numpy/core/tests/test_longdouble.py]: Heh, I didn't expect that, I would have expected an error. I assume what is being checked is that `,` is not handled as the radix character?

Comment (aarchiba) [inline numpy/core/tests/test_longdouble.py]: That's how fromstring normally "handles" invalid input. But if , is the decimal marker in your locale and you're in python 2.6, as @pv points out, you don't get the 1 either.

**Comment by charris** [comment]
Thanks Anne. I notice you don't have commit rights, they got lost in the move to github. Do you want them back?

**Comment by aarchiba** [comment]
The pull request workflow makes them a lot less necessary, and I think I need to get more used to the numpy coding style before I could approve any, but I'm willing to help when I have time. So if you think it would be useful, sure.

**Comment by charris** [comment]
I set you an invitation. If nothing else you are on the growing developers emeritus team;)

**Pull request merged**